File: .examples/.kt
class Person(val name: String, val age: Int) {
    fun greet(): String {
        return "Hello, my name is $name and I am $age years old."
    }
}

fun main() {
    val currentYear = 2023
    val personName = "Alice"
    val personAge = 30

    val person = Person(personName, personAge)
    val yearOfBirth = currentYear - personAge

    println(person.greet())
    println("${person.name} was born in $yearOfBirth.")
}

File: .examples/.rs
struct Person {
  name: String,
  age: u32,
}

impl Person {
  fn new(name: &str, age: u32) -> Self {
    Self {
      name: name.to_string(),
      age,
    }
  }

  fn greet(&self) -> String {
    format!("Hello, my name is {} and I am {} years old.", self.name, self.age)
  }
}

fn main() {
  let current_year = 2023;
  let person_name = "Alice";
  let person_age = 30;
  let year_of_birth = current_year - person_age;
  let person = Person::new(person_name, person_age);

  println!("{}", person.greet());
  println!("{} was born in {}.", person.name, year_of_birth);
}
